transports/p2p/connmgr/connmanager.go

```diff
@@ -272,7 +323,10 @@ out:
 				conns[connReq.id] = connReq
 				cm.log.Debug().Msgf("Connected to %v", connReq)
 				connReq.retryCount = 0
-				cm.failedAttempts = 0
+				cm.resetGlobalFailedAttempts()
+				if connReq.Addr != nil && connReq.Addr.String() != "" {
+					cm.failedAttempts[connReq.Addr.String()] = 0
+				}
```

Comment:
### Matter of taste
I would probably extract it into something like:
```suggestion
				cm.resetFailedAttempts(connReq)
```
And inside we can reset global and address failed attempts

Reply:
Good point, I changed it so I could actually reuse a mutex and remove one redundant method